airflow/contrib/hooks/gcs_hook.py

```diff
@@ -655,6 +655,54 @@ def insert_object_acl(self, bucket, object_name, entity, role, generation,
                 'Object ACL entry creation failed. Error was: {}'.format(ex.content)
             )
 
+    def compose(self, bucket, source_objects, destination_object):
+        """
+        Composes a list of existing object into a new object in the same storage bucket
+
+        Currently it only supports up to 32 objects that can be concatenated
+        in a single operation
+
+        https://cloud.google.com/storage/docs/json_api/v1/objects/compose
+
+        :param bucket: The name of the bucket containing the source objects.
+            This is also the same bucket to store the composed destination object.
+        :type bucket: str
+        :param source_objects: The list of source objects that will be composed
+            into a single object.
+        :type source_objects: list
+        :param destination_object: The path of the object if given.
+        :type destination_object: str
+        """
+
+        if not source_objects or not len(source_objects):
+            raise ValueError('source_objects cannot be empty.')
+
+        if not bucket or not destination_object:
+            raise ValueError('bucket and destination_object cannot be empty.')
+
+        service = self.get_conn()
+
+        dict_source_objects = [{'name': source_object}
+                               for source_object in source_objects]
+        body = {
+            'sourceObjects': dict_source_objects
+        }
+
+        try:
+            self.log.info("Composing %s to %s in the bucket %s",
+                          source_objects, destination_object, bucket)
+            service \
+                .objects() \
+                .compose(destinationBucket=bucket,
+                         destinationObject=destination_object,
+                         body=body) \
+                .execute()
```

Comment:
```suggestion
                .execute(min_retries=5)
```

Reply:
You are looking at bad documentation. There are currently two libraries for Google Cloud Storage. Old based on discovery (`google-api-python-client`). New is native (`google-cloud-storage`). Documentation for the old library disappears from the internet, but there is a short note in one place.
https://googleapis.github.io/google-api-python-client/docs/epy/googleapiclient.http.HttpRequest-class.html#execute
Note that most other operators set this parameter.
Others operators will do it in the future when I finish my PRs: https://github.com/PolideaInternal/airflow/pull/50

BTW. If you want to develop operators for GCP, I recommend reading also:
https://docs.google.com/document/d/1_rTdJSLCt0eyrAylmmgYc3yZr-_h51fVlnvMmWqhCkY/edit

Reply:
Done. Just committed this.